### PR TITLE
FIX: invite users with sufficient trust level

### PR DESCRIPTION
**Pull request by jahan-ggn**
Adding a PR for https://meta.discourse.org/t/please-make-tl-for-sending-invites-customisable/139828/18

### CODE REVIEW

File: spec/components/guardian_spec.rb [hunk 1]
```diff
@@ -610,7 +625,8 @@
     end
 
     describe "private messages" do
-      fab!(:user) { Fabricate(:user, trust_level: TrustLevel[2]) }
+      SiteSetting.min_trust_level_to_allow_invite = 2
```
Comment (eviltrout) [inline spec/components/guardian_spec.rb:628]: I am surprised this works, as this code is not evaluated during the test run. This statement should be moved inside the `before` below.

File: config/locales/server.en.yml [hunk 2]
```diff
@@ -1814,6 +1814,7 @@ en:
     min_trust_to_post_embedded_media: "The minimum trust level required to embed media items in a post"
     min_trust_level_to_allow_profile_background: "The minimum trust level required to upload a profile background"
     min_trust_level_to_allow_user_card_background: "The minimum trust level required to upload a user card background"
+    min_trust_level_to_allow_invite: "The minimum trust level required to allow invite"
```
Comment (eviltrout) [inline config/locales/server.en.yml:1817]:
```suggestion
    min_trust_level_to_allow_invite: "The minimum trust level required to invite users"
```

**Review by eviltrout — COMMENTED** [review]
Overall this looks great. I've made a couple suggestions.

File: spec/components/guardian_spec.rb (hunk 1 above)
Comment (jahan-ggn) [reply spec/components/guardian_spec.rb]: @eviltrout said changes are done

File: config/locales/server.en.yml (hunk 2 above)
Comment (jahan-ggn) [reply config/locales/server.en.yml]: @eviltrout said changes are done

File: config/site_settings.yml [hunk 3]
```diff
@@ -1406,6 +1406,10 @@ trust:
     default: 0
     client: true
     enum: "TrustLevelSetting"
+  min_trust_level_to_allow_invite:
+    default: 2
+    client: true
```
Comment (eviltrout) [inline config/site_settings.yml:1411]: I just saw this is a client setting but doesn't seem to be used on the client side. I think we should remove `client: true` here.

**Review by eviltrout — CHANGES_REQUESTED** [review]
This is looking good. I think after we remove the client thing it will be good to merge.

File: config/site_settings.yml (hunk 3 above)
Comment (jahan-ggn) [reply config/site_settings.yml:1411]: @eviltrout said changes are done

**Review by eviltrout — APPROVED** [review]

**Comment by eviltrout** [comment]
Thanks for this! It's a nice change.

**Comment by jahan-ggn** [comment]
@eviltrout is there any problem with discourse badges?
Because i have contributed 3 pr's and all 3 of them are merged by you but i am not getting the contributor badge
Please do have a look

**Comment by eviltrout** [comment]
@jahan-ggn that badge is only automatic if we can link your meta account to github via email. If you have your email private on github (looks like you do) we can't do it automatically. I've manually granted you the badge though!
Thanks again.

**Comment by jahan-ggn** [comment]
Thank you @eviltrout

**Pull request merged**